src/main/scala/ir/dsl/DSL.scala

```
@@ -164,21 +167,21 @@ case class EventuallyCall(
 case class EventuallyGoto(targets: Iterable[DelayNameResolve], label: Option[String] = None)
     extends EventuallyJump
     with DefaultDeepEquality {
-  override def resolve(p: Program, proc: Procedure): GoTo = {
-    val tgs = targets.flatMap(tn => tn.resolveBlock(p))
+  override def resolve(p: Resolver, proc: String): GoTo = {
+    val tgs = targets.flatMap(tn => tn.resolveBlock(p, proc))
     GoTo(tgs, label)
```

i know this isn't new in this PR, but a flatMap will silently discard targets which don't resolve successfully. is this a good idea?

---

Yeah I should fix this